### ISSUE TITLE
Cannot move item to another character when there's 1 slot left in all of inventory

### ISSUE THREAD

**Issue by nev-r** [DestinyItemManager/DIM#8244]
dim cannot solve this move
![image](https://user-images.githubusercontent.com/68782081/160259769-bf075fbc-f8e4-4bf8-a7ed-5eec6a799b10.png)

but it can solve this move
![image](https://user-images.githubusercontent.com/68782081/160259790-0a6434a7-3450-47ea-b406-662bea321a24.png)

**Comment by bhollis** [comment]
Interesting!

**Comment by bhollis** [comment]
I assume all your other buckets are also full?

**Comment by nev-r** [comment]
yeah these are the last two empty spaces in this case

**Comment by bhollis** [comment]
My guess is it's reserving a space for the item to move "through" that it doesn't actually need.

**Comment by ahydronous** [comment]
Just to chime in, but isn't this always how its been? Whenever I move stuff via DIM, it first always shuffles it into my vault before transferring it to the other character.

**Comment by bhollis** [comment]
That's not what this is talking about. DIM will generally move items to vault in order to make space for items you move.

**Comment by FlaminSarge** [comment]
This _seems_ to be closed by #10651; I've set up the situation where I have 1 open slot on 2 characters each with 700/700 vault and 49/50 consumables (the move refuses to attempt to fill the 50th slot, which I believe is intended); moving the currently-equipped item from one character to another seems to work now. However, I noticed that prior to the above patch, neither of the two initially reported cases work, as both balk at trying to move an item out of vault to make room for the item coming from the source character. Post-patch, they don't run into this issue.
There is one remaining bug, which is that if there is only ONE empty space available and it's in the same bucket/char as the item being moved, DIM refuses to fill that spot to make room in vault even though it's the source bucket and would be fine being filled. This is documented in #10651's description.
